src/pybind/mgr/rook/module.py

```diff
@@ -392,6 +392,14 @@ def remove_stateless_service(self, service_type, service_id):
             lambda: self.rook_cluster.rm_service(service_type, service_id), None,
             "Removing {0} services for {1}".format(service_type, service_id))
 
+    def update_mons(self, num, hosts):
+        if hosts:
+            raise RuntimeError("Host list is not supported by rook.")
```

Comment:
I think that this behavior makes sense. Ultimately I think these nuances should just be documented for each orchestrator backend. For example the SSH orchestrator throws an exception if you don't provide the exact number of hosts, but it could just as easily have ignored extra hosts...

Reply:
The user might want to give hints about where to install the mons. I'm imagining the sequence:
- If hosts is set, set a predefined label on each of the K8s nodes (such as `role=mon-node`)
- Update the cluster CRD with the number of mons and an updated `placement` for the mons to pick up the label.

Although some might argue that rook shouldn't be setting labels on nodes

Reply:
I was thinking that Rook completely hides host management from the orchestrator layer, so this makes sense.

Reply:
Ok, I could see that being doable. So basically doing:

`ceph orchestrator mon update 3 foo.bar.baz,baz.bar.foo`

...would make it so that you'd update the count to 3 mons, and make those two hosts be the only ones on the allowed list.

Reply:
something like that. Although, I'd make sure that `len(hosts) == num`

Reply:
Ok. This proposal is a bit more complex than what we have currently in any backend. I think I'd like to see a bit more concrete info about how we'd implement this.

@travisn , could you paste in a sample placement: section from such a CRD? The rook docs for it are a bit skimpy and it seems rather complex.

Reply:
Here is an example of the placement yaml:
```yaml
  placement:
    mon:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: role
              operator: In
              values:
              - mon-node
```

I would also want to check that `len(hosts) >= num`. If there are extra hosts, the operator would have flexibility to failover.

This is likely better as a separate PR since it does complicate things.

Reply:
Agreed. Can I suggest that we take this PR as-is, and look at doing that at some point in the future? I too am not sure I like the idea of rook setting labels on nodes.